hack/bundle-gen.py

```diff
@@ -200,6 +200,60 @@ def build_and_push_image(registry_auth_file, image_repo, image_tag, dry_run, bui
     )
 
 
+# get_previous_version grabs the previous hive version (without the leading `v`) from
+# COMMUNITY_OPERATORS_HIVE_PKG_URL package yaml for the provided channel_name.
+def get_previous_version2(work_dir, channel_name):
+    upstream_repo = "git@github.com:dlom/community-operators-prod.git"
+    upstream_branch = "mold-convert-to-bundle-test-branch"
+    dir_name = "previous-version-determiner"
+
+    repo_full_path = os.path.join(work_dir, dir_name)
+    # clone git repo
+    try:
+        git.Repo.clone_from(upstream_repo, repo_full_path)
+    except:
+        print("Failed to clone repo {} to {}".format(upstream_repo, repo_full_path))
+        raise
+
+    # get to the right place on the filesystem
+    print("Working in %s" % repo_full_path)
+    os.chdir(repo_full_path)
+
+    repo = git.Repo(repo_full_path)
+
+    # Starting branch
+    print("Checkout latest {}".format(upstream_branch))
+    try:
+        repo.git.checkout(upstream_branch)
+    except:
+        print("Failed to checkout {}".format(upstream_branch))
+        raise
+
+    try:
+        versions = []
+        hive_dir = os.path.join(repo_full_path, HIVE_SUB_DIR)
+        for version in os.listdir(hive_dir):
+            annotation_yaml_path = os.path.join(hive_dir, version, "metadata", "annotations.yaml")
+            with open(annotation_yaml_path, "r") as stream:
+                annotation_yaml = yaml.load(stream, Loader=yaml.SafeLoader)
+                version_channel = annotation_yaml["annotations"]["operators.operatorframework.io.bundle.channel.default.v1"]
+                if version_channel == channel:
+                    versions.append(version)
+    except:
+        print(
+            "Unable to determine previous hive version from {}",
+            upstream_repo,
+        )
+        raise
+
+    versions.sort(reverse=True)
+    if len(versions) == 0:
+        # Channel not found -- no previous version.
+        return None
+
+    return versions[0]
```

Comment:
Couple things here:
- We shouldn't use lexicographic sort, as e.g. 1.2.100-* will be "less than" 1.2.99-*. Recommend using [semver](https://pypi.org/project/semver/), specifically [compare](https://python-semver.readthedocs.io/en/stable/usage.html#comparing-versions).
- There's really no need to save all the versions in a list -- just save the largest version seen so far.